api/src/main/java/org/pmiops/workbench/billing/BillingProjectBufferService.java

```diff
@@ -56,19 +60,28 @@ public void bufferBillingProject() {
 
   public void syncBillingProjectStatus() {
     BillingProjectBufferEntry entry = billingProjectBufferEntryDao
-        .findFirstByStatusOrderByCreationTimeAsc(StorageEnums.billingProjectBufferStatusToStorage(CREATING));
+        .findFirstByStatusOrderByLastSyncRequestTimeAsc(StorageEnums.billingProjectBufferStatusToStorage(CREATING));
 
     if (entry == null) {
       return;
     }
 
-    switch (fireCloudService.getBillingProjectStatus(entry.getFireCloudProjectName()).getCreationStatus()) {
-      case READY:
-        entry.setStatusEnum(AVAILABLE);
-        break;
-      case ERROR:
-        entry.setStatusEnum(ERROR);
-        break;
+    entry.setLastSyncRequestTime(new Timestamp(clock.instant().toEpochMilli()));
+
+    try {
+      switch (fireCloudService.getBillingProjectStatus(entry.getFireCloudProjectName())
+          .getCreationStatus()) {
+        case READY:
+          entry.setStatusEnum(AVAILABLE);
+          break;
+        case ERROR:
+          log.warning(String.format("SyncBillingProjectStatus: BillingProject %s creation failed",
+              entry.getFireCloudProjectName()));
+          entry.setStatusEnum(ERROR);
+          break;
+      }
+    } catch (Exception e) {
```

Comment:
I'd make this a more specific type to avoid suppressing programming errors - are you trying to catch a FirecloudApiException here?

Reply:
yeah. It doesn't look like Firecloud. ApiException gets thrown though.

It seems like FireCloudService always returns a WorkbenchException so I catch that instead.